TableToMarkdown docs (#1293)
demisto/content-docs: docs/integrations/code-conventions.md

In the War Room, this is how a table will appear:
<img width="788" src="/doc_imgs/integrations/50571324-46846e00-0db0-11e9-9888-ddd9dc275541.png"></img>
#
You may also use ```headerTransform``` to convert the existing keys into formatted headers.

> A function that formats the original data headers (optional).

Reviewer: ```suggestion
|This function formats the original data headers (optional).
```

```python
t = {'header_1': 'a1', 'header_2': 'b1', 'header_3': 'c1'}

> tableToMarkdown('tableToMarkdown with headerTransform', t, headerTransform=underscoreToCamelCase)
> ```
> The above will create the table seen below:
> ### tableToMarkdown with headerTransform

Reviewer 2: ```suggestion
tableToMarkdown('headerTransform Example', t, headerTransform=underscoreToCamelCase)
```
The above will create the table seen below:
### headerTransform Example
```

|Header1|Header2|Header3|
|---|---|---|
| a1 | b1 | c1 |
#
You may also use ```removeNull``` to remove empty columns in the table. Default is False.
```python
headers = ['header_1', 'header_2']
data = {
    'header_1': 'foo',
}

> tableToMarkdown('tableToMarkdown test', data, removeNull=True, headers=headers)
> ```
> The above will create the table seen below:
> ### tableToMarkdown test

Reviewer 2: ```suggestion
tableToMarkdown('removeNull Example', data, removeNull=True, headers=headers)
```
The above will create the table seen below:
### removeNull Example
```

The resulted table will be:

![image](https://user-images.githubusercontent.com/72340690/103922604-a25efb80-511c-11eb-9021-c062226b5001.png)
#
Use the ```date_fields``` argument (list) of date fields to format the value to human-readable output.
```python
data = [
    {
        "docker_image": "demisto/python3",
        "create_time": '1631521313466'
    }
]

> tableToMarkdown('tableToMarkdown test', data, headers=["docker_image", "create_time"],
>                 date_fields=['create_time'])
> ```
> The above will create the table seen below:
> ### tableToMarkdown test
> |docker_image|create_time|

Reviewer 2: ```suggestion
tableToMarkdown('tableToMarkdown date_fields example', data, headers=["docker_image", "create_time"],
                date_fields=['create_time'])
```
The above will create the table seen below:
### tableToMarkdown date_fields example
|docker_image|create_time|
```

|---|---|
| demisto/python3 | 2021-09-13 08:21:53 |

#

> Use the ```json_transform_mapping``` argument (Dict[str, JsonTransformer]), mapping between a header key to corresponding JsonTransformer.

Reviewer: ```suggestion
Use the ```json_transform_mapping``` argument (Dict[str, JsonTransformer]), to map between a header key to the corresponding JsonTransformer.
```

```python
data_with_list = {
  "Machine Action Id": "5b38733b-ed80-47be-b892-f2ffb52593fd",
  "MachineId": "f70f9fe6b29cd9511652434919c6530618f06606",
  "Hostname": "desktop-s2455r9",
  "Status": "Succeeded",
  "Creation time": "2022-02-17T08:20:02.6180466Z",
  "Commands": [
    {
      "startTime": null,
      "endTime": "2022-02-17T08:22:33.823Z",
      "commandStatus": "Completed",
      "errors": ["error1", "error2", "error3"],
      "command": {
        "type": "GetFile",
        "params": [
          {
            "key": "Path",
            "value": "test.txt"
          }
        ]
      }
    },
    {
      "startTime": null,
      "endTime": "2022-02-17T08:22:33.823Z",
      "commandStatus": "Completed",
      "errors": [],
      "command": {
        "type": "GetFile",
        "params": [
          {
            "key": "Path",
            "value": "test222.txt"
          }
        ]
      }
    }
  ]
}
table = tableToMarkdown("tableToMarkdown test", data_with_list,
            json_transform_mapping={'Commands': JsonTransformer(keys=('commandStatus', 'command'))})
```
The above will create the table seen below:
![img_1.png](img_1.png)
#

> Use the ```is_auto_json_transform``` argument (bool), to try to auto transform complex json.

Reviewer: ```suggestion
Use the ```is_auto_json_transform``` argument (bool), to try to auto transform a complex JSON.
```
